zmq-deferred/src/socket.ml

```diff
@@ -132,7 +132,12 @@ module Make(T: Deferred.T) = struct
   let to_socket t = t.socket
 
   let recv s = post s Receive (fun s -> ZMQ.Socket.recv ~block:false s)
-  let send s m = post s Send (fun s -> ZMQ.Socket.send ~block:false s m)
+  let send ?more s m =
+    post s Send (fun s -> ZMQ.Socket.send ?more ~block:false s m)
```

Comment:
I think this is dangerous. If there are multiple writers, then we will have strange interleaving between fragments and messages. 
I would advice that users only use the send_all / send_msg_all functions. We could also add call that takes a message producer something like
`val send_more: t -> (unit -> string option Deferred.t) -> unit` that would read message fragments from the given function until the function returned None. Same pattern could be used for a recv_more function.

Reply:
Fair point.  I think there's value in this as it was something a user created a PR for in the past.  I can pull this change out for now though and we can figure out the right way to do this in a separate PR.

Reply:
Removed in 93bf573

Reply:
I can be persuaded, but I don't like it. 
At least there should be a comment discouraging users from using it.

Reply:
You choose. I don't like breaking other peoples code either.

Reply:
I like putting it in a separate PR/discussion.  I agree that it should be discouraged in the general case.